pkg/controllers/nodeclass/status/instance_profile/controller.go

```diff
@@ -77,36 +59,40 @@ func (c *Controller) Reconcile(ctx context.Context, nodeClass *v1beta1.EC2NodeCl
 		}
 	}
 	stored := nodeClass.DeepCopy()
-
-	var results []reconcile.Result
 	var errs error
-	for _, reconciler := range []nodeClassStatusReconciler{
-		c.ami,
-		c.subnet,
-		c.securitygroup,
-		c.instanceprofile,
-		c.readiness,
-	} {
-		res, err := reconciler.Reconcile(ctx, nodeClass)
-		errs = multierr.Append(errs, err)
-		results = append(results, res)
+	if nodeClass.Spec.Role != "" {
```

Comment:
nit: IMO it would be cleaner to pull this logic out into a helper function, and do away with the multierror. This is going to be a theme in my review, but we should avoid multierrors if they're not needed since it's use implies there will be multiple errors IMO. 

```go
func (c *Controller) Reconcile(...) {
    ...
    instanceProfile, err := c.resolveInstanceProfile(nodeClass)
    if err != nil {
        nodeClass.StatusConditions().SetFalse(v1beta1.ConditionTypeInstanceProfileReady, "InstanceProfileCreateError", "Error creating instance profile")
    } else {
        nodeClass.Status.InstanceProfile = instanceProfile
        nodeClass.StatusConditions().SetTrue(v1beta1.ConditionTypeInstanceProfileReady)
    }

    if !equality.Semantic.DeepEqual(stored, nodeClass) {
        ...
    }
    if err != nil {
        return reconcile.Result{}, err
    }
    ...
}

func (c *Controller) resolveInstanceProfile(nc *v1beta1.EC2NodeClass) (string, error) { 
    if nodeClass.Spec.Role == "" {
        return lo.FromPtr(nodeClass.Spec.InstanceProfile), nil
    }
    return c.instanceProfileProvider.Create(ctx, nc)
}
```